### PR TITLE
Adding manifests for kfserving

### PR DESCRIPTION
**Pull request by johnugeorge**
**Description of your changes:**
This PR contains manifests for knative release 0.8 and kfserving 0.1.1. This also adds service role entry that gives access to all services in knative-serving namespace since istio cluster rbac is ON by default.

Related: https://github.com/kubeflow/kubeflow/issues/3624
/hold

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/268)

### CODE REVIEW

File: kfserving/kfserving-install/base/service.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    controller-tools.k8s.io: "1.0"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    control-plane: kfserving-controller-manager
+    controller-tools.k8s.io: "1.0"
+  name: kfserving-controller-manager-service
+  namespace: kfserving-system
```
Comment (kkasravi) [inline kfserving/kfserving-install/base/service.yaml]: namespace not needed

File: kfserving/kfserving-install/base/service.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+metadata:
+  annotations:
+    prometheus.io/port: "8443"
+    prometheus.io/scheme: https
+    prometheus.io/scrape: "true"
+  labels:
+    control-plane: controller-manager
+    controller-tools.k8s.io: "1.0"
+  name: kfserving-controller-manager-metrics-service
+  namespace: kfserving-system
```
Comment (kkasravi) [inline kfserving/kfserving-install/base/service.yaml]: namespace not needed

File: kfserving/kfserving-install/base/config-map.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            "image": "gcr.io/kfserving/xgbserver"
+        },
+        "tensorrt": {
+            "image": "nvcr.io/nvidia/tensorrtserver"
+        }
+    }
+kind: ConfigMap
+metadata:
+  name: kfservice-config
+  namespace: kfserving-system
```
Comment (kkasravi) [inline kfserving/kfserving-install/base/config-map.yaml]: if you do something like the following
```
  - kustomizeConfig:
      parameters:
      - name: namespace
        value: kfserving-system
      repoRef:
        name: manifests
        path: kfserving/kfserving-install
    name: kfserving-install
```

in a config file that deploys kfserving (see [kfctl_gcp_iap.yaml](https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/kfctl_gcp_iap.yaml) for example)
then you can remove all namespace lines including I believe

```
subjects:
- kind: ServiceAccount
  name: default
  namespace: kfserving-system
```

File: kfserving/kfserving-install/base/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  labels:
+    control-plane: kfserving-controller-manager
+    controller-tools.k8s.io: "1.0"
+  name: kfserving-controller-manager
+  namespace: kfserving-system
```
Comment (kkasravi) [inline kfserving/kfserving-install/base/statefulset.yaml]: namespace not needed

File: kfserving/kfserving-install/base/statefulset.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        - mountPath: /tmp/cert
+          name: cert
+          readOnly: true
+      terminationGracePeriodSeconds: 10
+      volumes:
+      - name: cert
+        secret:
+          defaultMode: 420
+          secretName: kfserving-webhook-server-secret
+  volumeClaimTemplates: []
```
Comment (kkasravi) [inline kfserving/kfserving-install/base/statefulset.yaml]: we won't need this hack when we upgrade kustomize which is at v2.0.3

File: knative/knative-install/base/cluster-role-binding.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    serving.knative.dev/release: "v0.8.0"
+  name: custom-metrics:system:auth-delegator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+- kind: ServiceAccount
+  name: controller
+  namespace: knative-serving
```
Comment (kkasravi) [inline knative/knative-install/base/cluster-role-binding.yaml]: i believe you can remove this namespace but you should confirm

**Comment by kkasravi** [comment]
Thanks - I would like to run it, and will do so in the next few hours.

**Comment by johnugeorge** [comment]
Thanks @kkasravi

**Comment by jlewi** [comment]
/lgtm
/hold
Placing a hold so @kkasravi can finish his review

**Comment by kkasravi** [comment]
Hi @johnugeorge 

see #286 - i added a registry var to your PR and also rebased from master to pull in the latest unit-tests. If you pull #286 into this PR you should be good to go. 
Files modified/added:
```
manifests/kfserving/kfserving-install/base/params.env
manifests/kfserving/kfserving-install/base/params.env
manifests/kfserving/kfserving-install/base/kustomization.yaml
manifests/kfserving/kfserving-install/base/config-map.yaml
manifests/kfserving/kfserving-install/base/statefulset.yaml
```

**Comment by kkasravi** [comment]
@googlebot I consent.

**Comment by johnugeorge** [comment]
Removing the hold as @kkasravi finished his review.
/hold cancel

**Pull request merged**